common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::dcs_scan_trigger(int dwell_time_msec,
+                                          const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams dcs_scan_params;
+    memset((void *)&dcs_scan_params, 0, sizeof(dcs_scan_params)); //must to ensure freq are zero
+    size_t out_len                                = 0;
+    unsigned char out_data[NL_MAX_REPLY_BUFFSIZE] = {'\0'};
+    sScanCfgParams fg, new_fg;   //foreground scan param
+    sScanCfgParamsBG bg, new_bg; //background scan param
+
+    auto ifname = m_radio_info.iface_name;
+
+    DWPAL_GET_SCAN_PARAMS_HELPER_FG(ifname, out_data, out_len, NL_MAX_REPLY_BUFFSIZE, fg);
```

Comment:
Should use static inline functions and check return value.

Reply:
can't use static, they use protected methods called "dwpal_nl_cmd_get" and "dwpal_nl_cmd_set"

Reply:
You can define static inline private methods in the header.

Reply:
but since the cmd get & set are protected member methods they need an instance of the base_wlan_hal_dwpal object (in this case the local "this" is mon_wlan_hal_dwpal which inherits it) to use the two functions

so if we want to remove the macros we have to either:
1) make static functions that will receive an reference to base_wlan_hal_dwpal, and in the function call will add "this" as the handler
2) create local member methods that wrap the protected functions, they will use additional utility we gain from the macros